src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Android/StorePal.cs

```diff
@@ -44,7 +44,47 @@ public static IExportPal LinkFromCertificateCollection(X509Certificate2Collectio
 
         public static IStorePal FromSystemStore(string storeName, StoreLocation storeLocation, OpenFlags openFlags)
         {
-            throw new NotImplementedException(nameof(FromSystemStore));
+            bool isReadWrite = (openFlags & OpenFlags.ReadWrite) == OpenFlags.ReadWrite;
+            if (isReadWrite && storeLocation == StoreLocation.LocalMachine)
+            {
+                // All LocalMachine stores are read-only
+                throw new CryptographicException(
+                    SR.Cryptography_Unix_X509_MachineStoresReadOnly,
+                    new PlatformNotSupportedException(SR.Cryptography_Unix_X509_MachineStoresReadOnly));
+            }
+
+            StringComparer ordinalIgnoreCase = StringComparer.OrdinalIgnoreCase;
+            switch (storeLocation)
+            {
+                case StoreLocation.CurrentUser:
+                {
+                    // Matches Unix behaviour of getting a disallowed store that is always empty.
```

Comment:
I haven't dealt with the `My` store in this change.

From what I can tell, when Android was using OpenSSL directly, the `My` store for `CurrentUser` was really for the current application (started out empty, not shared with other applications, removed on app uninstall/reinstall). I can match that behaviour and `My`+`CurrentUser` on Android will really mean current application?

Reply:
CurrentUser\My should be wherever a normal Java application would put/get certificates it uses.  That... might... be the "PKCS12" KeyStore?  Or the "give me the default KeyStore" KeyStore?  Whatever is "natural".

Reply:
e.g. on macOS CurrentUser\My is the user's default keychain (generally login.keychain, but we ask for the default, not any keychain by name, and it can be replaced).

Reply:
I guess current application (`AndroidKeyStore`) might actually be the 'natural' thing for Android apps, since they are all sandboxed and running under their own user ID.

Reply:
Directed CurrentUser\My to use the store for the current application - currently just throwing `NotImplementedException` if it tries to add any certs with a private key.